### PR TITLE
Refactor chef 13 style

### PR DESCRIPTION
**Pull request by meringu**
This change refactors the resources to modernize them. Other change is to install the packages directly instead of with the AWS provided installer script. This change should also fix the failing builds in master.

Closes #5

### CODE REVIEW

File: libraries/helper.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      file_name = ::File.basename(version_hash[package_type])
+      "#{Chef::Config[:file_cache_path]}/code_deploy/#{file_name}"
+    end
+
+    # If the machine uses systemd
+    #
+    # @return [Boolean]
+    #
+    def systemd?
+      cmd = shell_out('systemctl | grep "\-\.mount"')
```
Comment (zl4bv) [inline libraries/helper.rb]: There's a helper in Chef core that may be more reliable at determining if the node uses systemd

```ruby
Chef::Platform::ServiceHelpers.service_resource_providers.include?(:systemd)
```

**Comment by zl4bv** [comment]
Do the CodeDeploy packages come with signatures that can be verified?

File: test/integration/default/inspec/code_deploy_spec.rb [hunk 2]
```diff
@@ -15,4 +15,22 @@
   describe file('/etc/cron.d/codedeploy-agent-update') do
     it { should_not exist }
   end
+
+  describe file('/etc/codedeploy-agent/conf/codedeployagent.yml') do
```
Comment (zl4bv) [inline test/integration/default/inspec/code_deploy_spec.rb]: Consider using the [`yaml`](https://www.inspec.io/docs/reference/resources/yaml/) resource.

**Comment by meringu** [comment]
> Do the CodeDeploy packages come with signatures that can be verified?

I can't see any info in the docs about the packages being signed http://docs.aws.amazon.com/codedeploy/latest/userguide/codedeploy-agent-operations-install.html. However `remote_file` should check the etag to at least check the file isn't corrupt.

**Review by zl4bv — CHANGES_REQUESTED** [review]

**Review by zl4bv — APPROVED** [review]

**Pull request merged**